src/cluster/placement/selector/common.go

```diff
@@ -46,6 +46,10 @@ func getValidCandidates(
 		}
 		if instanceInPlacement.IsLeaving() {
 			instances = append(instances, instanceInPlacement)
+			continue
+		}
+		if opts.ShardStateMode() == placement.StableShardStateOnly {
```

Comment:
Talked offline, going to remove this check for now until we have a more concrete use case for it.